### DISCUSSION TITLE
I think we should move metamask and coinbase connector into injected

### DISCUSSION THREAD

**Discussion by chendatony31** [wevm/wagmi#2177] [General]
As the code in  https://github.com/wagmi-dev/references/tree/main/packages/connectors/src
I think most injected wallet connectors are the same besides the name. I don't know why only metamask and coinbase are separated.  I think we should move them into the injected.  Just like [`web3-onboard`](https://github.com/blocknative/web3-onboard) do.

**Comment by jxom** [comment]
They are separated as the MetaMask and Coinbase extensions themselves have behaviour which other injected wallets don’t have (ie. shimming account selection, disconnection, request perms, etc). We simply can’t trust that all wallets that impersonate MetaMask will have the _exact_ same implementation and behaviour as MetaMask (especially the security concerns around request permissions), hence why there is a Connector to target MetaMask as well as options to work around its nuances. The Coinbase Connector also has compatibility for Coinbase mobile through their SDK too (which we cannot do via injected).

IMO, once wallets start to implement [EIP-5749](https://eips.ethereum.org/EIPS/eip-5749) to support multiple injected providers, this will dramatically improve the UX and DX of selecting injected wallets as we won’t have to worry about wallet impersonation.

**Reply by tmm** [reply]
Also, the Coinbase Wallet Connector uses an injected provider in browsers and QR code system when the injected provider is not installed.

**Reply by chendatony31** [reply]
So if we have some Interface or function that is different from an injected wallet, we can add another connector?

**Reply by jxom** [reply]
Yes. Read more: https://github.com/wagmi-dev/references/blob/main/.github/CONTRIBUTING.md#connectors

**Reply by chendatony31** [reply]
OK, thanks, please have a look at this PR: https://github.com/wagmi-dev/references/pull/229
